src/mqtt_client.rs

```diff
@@ -305,15 +301,18 @@ where
 
             // Note(unwrap): This should not fail because `can_publish()` was checked before
             // attempting this publish.
-            self.mqtt
-                .client()
-                .publish(
-                    Publication::new(&data[..len])
-                        .topic(&prefixed_topic)
-                        .finish()
-                        .unwrap(),
-                )
-                .unwrap();
+            match self.mqtt.client().publish(
+                // If the topic is not present, we'll fail to serialize the setting into the
+                // payload and will never publish. The iterator has already incremented, so this is
+                // acceptable.
+                DeferredPublication::new(|buf| self.settings.get_json(&topic, buf))
+                    .topic(&prefixed_topic)
+                    .finish()
+                    .unwrap(),
+            ) {
+                Err(minimq::PubError::Serialization(Error::Absent(_))) => {}
+                other => other.unwrap(),
```

Comment:
Should we not catch `miniconf::Error::Inner()`, i.e. buf too short instead of panicing?

Reply:
It's not clear to me what handling that issue would actually do. I don't think we want to ignore those issues and continue operating, as the user of miniconf would then silently have potential settings that just aren't getting published and/or cannot be set.

What would we do instead of panicking here?

Reply:
If we handle this message and log it, I have concerns. Specifically:
1. Logging is not required on the end-user side, so they may be unaware of the issue
2. A silently unavailable settings seems very problematic

I honestly think an internal panic here is appropriate because it immediately informs the developer at the development stage to fix the issue and doesn't allow it to potentially seep into a production build.

Reply:
I agree that for development a panic is very desirable here.
But for production, the panic is worse than most other options, especially if we know what we can do.
What about instead sending a message to that republication topic, either empty, with some error property attached, or with "<serialization failure>" as payload? Then there is feedback that something is wrong.

Reply:
I think those are both good ideas, I'll take a look at what makes the most sense.